Hazelcast.Test/Hazelcast.Client.Test/PagingPrediateTest.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) 2008, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Hazelcast.Client.Model;
+using Hazelcast.Config;
+using Hazelcast.Core;
+using Hazelcast.Util;
+using NUnit.Framework;
+
+namespace Hazelcast.Client.Test
+{
+    [TestFixture]
+    public class PagingPrediateTest : SingleMemberBaseTest
+    {
+        [SetUp]
+        public void Init()
+        {
+            map = Client.GetMap<object, object>(TestSupport.RandomString());
+        }
+
+        [TearDown]
+        public static void Destroy()
+        {
+            map.Clear();
+        }
+
+        protected override void ConfigureClient(ClientConfig config)
+        {
+            base.ConfigureClient(config);
+            config.GetSerializationConfig()
+                .AddDataSerializableFactory(IdentifiedFactory.FactoryId, new IdentifiedFactory());
+        }
+
+        internal static IMap<object, object> map;
+
+        private void FillMap()
+        {
+            for (var i = 0; i < 10; i++)
+            {
+                map.Put("key-" + i, "value-" + i);
```

Comment:
May be we can also put an entry with different key but the same value, e.g. : ("key-30", "value-2") . values test may fail in this test due to this issue: https://github.com/hazelcast/hazelcast/issues/9968 but it may pass if the java and csharp hashcodes are equal for the same object.